### PR TITLE
WebXR update to make `requestAnimationFrame` more closely match the spec

### PR DESCRIPTION
**Pull request by syntheticmagus**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://immersive-web.github.io/webxr/#xrsession-interface
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by RaananW — APPROVED** [review]
This is correct according to the specs. Thanks!

File: types/webxr/index.d.ts [hunk 1]
```diff
@@ -227,7 +227,7 @@ export interface XRSession {
      * canceling the callback using cancelAnimationFrame(). This method is comparable
      * to the Window.requestAnimationFrame() method.
      */
-    requestAnimationFrame: XRFrameRequestCallback;
+    requestAnimationFrame: (callback: XRFrameRequestCallback) => number;
```
Comment (robrohan) [inline types/webxr/index.d.ts:230]: It's odd my local copy already has this change... so odd.

Can you add something like this to the test?

```
...
.then(rs => {
                space = rs as XRReferenceSpace;
...
                const loop: XRFrameRequestCallback = (now: number, frame: XRFrame) => { }
                const n = session.requestAnimationFrame(loop);
                session.cancelAnimationFrame(n);
...
```

So we can lock this in place a bit better?

**Review by robrohan — CHANGES_REQUESTED** [review]

File: types/webxr/index.d.ts (hunk 1 above)
Comment (syntheticmagus) [reply types/webxr/index.d.ts:230]: Added! I'm not sure what the etiquette is on marking conversations resolved (I know some repos prefer leaving that to the original commenter as a confirmation that the issue was truly addressed), so I'll leave this open and re-request review. Thanks!

**Review by robrohan — APPROVED** [review]

**Comment by syntheticmagus** [comment]
Ready to merge

**Pull request merged**